core/modules/comment/comment.module

```diff
@@ -135,7 +151,7 @@ function comment_field_extra_fields() {
   $return = array();
 
   foreach (node_type_get_types() as $type) {
-    if ($type->settings['comment_subject_field']) {
+    if ($type->settings['comment_subject_options'] = 1) {
```

Comment:
You assign the value `=` instead of comparison `==`?

Reply:
Use the constant here instead of integer.

Reply:
Done